pkgs/tools/misc/sanctity/default.nix

```diff
@@ -0,0 +1,25 @@
+{ lib, stdenv, rustPlatform, fetchFromGitea }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "sanctity";
+  version = "1.2.1";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "papojari";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-1bgk4HxWdefhNFeEQrg9Jvj3aMvzFfIJl+eYK/AgX5g=";
```

Comment:
If you update a version, you also need to update this.

Reply:
What is that sha256 tho? I couldn't find the documentation for `fetchFromGitea`. Also building the package didn't fail for me.

Reply:
`sha256` is just checksum for the content you fetched hence verifies the content is intended.
Since `sha256` is unique (at least for now can safely assume to be), nix *thinks* there is derivation in your `/nix/store` and that is what you wanted. And continues to build using the commit/revision `v0.7.3`. And you built v0.7.3 before so it doesn't fail. For more about this you can read manual 11.1 Caveats
Yeah, clearly lack of documentation for `fetchFromGitea` but all fetchers live in the directory `pkgs/build-support` 
so I find that fetcher there. 
It is good to use `nixpkgs-review` this package more or less assume ignores what's in `/nix/store`
So build failure is reproducable for you as well.